includes/class-sensei-teacher.php

```diff
@@ -1205,51 +1205,35 @@ public function get_learner_ids_for_courses_with_edit_permission() {
 			return [];
 		}
 
-		$learner_ids_for_courses_with_edit_permission = [];
+		$course_ids             = wp_list_pluck( $courses_with_edit_permission, 'ID' );
+		$course_ids_placeholder = implode( ',', array_fill( 0, count( $course_ids ), '%d' ) );
 
-		foreach ( $courses_with_edit_permission as $course ) {
-
-			$course_learner_ids = array();
-			$activity_comments  = Sensei_Utils::sensei_check_for_activity(
-				array(
-					'post_id' => $course->ID,
-					'type'    => 'sensei_course_status',
-					'field'   => 'user_id',
-				),
-				true
+		global $wpdb;
+		if ( Progress_Storage_Settings::is_tables_repository() ) {
+			// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Using a direct query for performance reasons.
+			$learner_ids = (array) $wpdb->get_col(
+				$wpdb->prepare(
+					"SELECT DISTINCT user_id
+					FROM {$wpdb->prefix}sensei_lms_progress
+					WHERE type = 'course'
+					AND post_id IN ( $course_ids_placeholder )", // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared, WordPress.DB.PreparedSQLPlaceholders.UnfinishedPrepare -- Placeholders created dynamically.
```

Comment:
Hi @m1r0 👋 I think we can replace this `$course_ids_placeholder` with just a `%1s`, so this will look like `AND post_id IN ( %1s )`.

And you can remove the $course_ids_placeholder calculation above too, we won't need it. Rather we'll just need a comma separated string, which can be done in the same line where we are doing the plucking -
`$course_ids = implode( ',', wp_list_pluck( $courses_with_edit_permission, 'ID' ) );`

The main benefit of this is, we will be using $wpdb->prepare in its expected way. wpdb->prepare expects that we won't do any manual interpolation here, rather placeholders and replaces will be used for everything.

Second benefit is, as phpcs tries to enforce the rule mention above, we're using an ignore comment here to override that. I think it's better to follow the rule here, so we won't need this ignore comment anymore.

Third little benefit is probably one less line of code :p

Maybe it'd be helpful if we know why we are using %1s instead of %s (I assume your generating the '%d,%d,%d' placeholder string, interpolating it, and then replacing it to was avoid this same reason). When %s is replaced by a string, it wraps the replaced string in quotes, so the IN query looks like `AND post_id IN ( '1,2,3' )` and so it doesn't work, because the quotation marks break the query. But %1s outputs the id string as it is without quotation marks like AND post_id IN ( 1,2,3 )`, so the query works without issue.

Hope this helps <3 LMKWYT!